moml/data/datasets.py

```
@@ -77,360 +167,841 @@ def __init__(
+            ValueError: If required columns are missing
         """
-        self.transform = transform
+        if not os.path.exists(data_path):
+            raise FileNotFoundError(f"Data file not found: {data_path}")
 
-        # Load data
-        self.df = pd.read_csv(data_path)
+        try:
+            df = pd.read_csv(data_path)
+        except Exception as e:
+            raise ValueError(f"Failed to load CSV file {data_path}: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)